modules/ROOT/pages/teradata-vantage-architecture-and-concepts.adoc

```diff
@@ -0,0 +1,120 @@
+= Teradata Vantage Architecture and Concepts
+:experimental:
+:page-author: Krutik Pathak
+:page-email: krutik.pathak@teradata.com
+:page-revdate: TBD
+:description: Teradata Vantage Architecture and Concepts.
+:keywords: data warehouses, teradata vantage architecture, teradata, vantage, Access Module Processors (AMP), Parsing Engines (PE), Massively Parallel Processing (MPP), Virtual Disk (Vdisks)
+:tabs:
+:dir: teradata-vantage-architecture-concepts
+
+== Overview
+
+This tutorial explains the architecture of Teradata Vantage and the underlying concepts. 
+
+Teradata's architecture is designed around the Massively Parallel Processing (MPP) architecture, which enables high-performance data processing and analytics. The major architectural components of Teradata include the Parsing Engine, Message Passing Layer (MPL), Access Module Processors (AMPs), and Virtual Disks (Vdisks).
+
+
+image::{dir}/teradata_architecture_major_components.png[Teradata Vantage Major Architectural Components, width=75%]
+
+== Teradata Vantage Architecture Components
+The Teradata Vantage architecture consists of below components.
+
+=== Parsing Engines (PE)
+When a SQL query is run in Teradata, it first reaches the Parsing Engine. The functions of the PE are:
+
+* Manage individual user sessions (up to 120). 
+* Parse and optimize the SQL queries. 
+* Check if the user has required privileges against the objects used in the SQL query. 
+* Check if the objects used in the SQL exist. 
+* Prepare the execution plan to execute the SQL query and pass it to AMP. 
+* Receives the response from the AMPs and sends it back to the requesting client. 
+
+=== Message Passing Layer (MPL)
+Message Passing Layer is a combination of Parallel Database Extension (PDE) and BYNET Software. It enables communication between Parsing Engine (PE) and Access Module Processors (AMP).  The primary function of MPL is to carry messages between AMPs and PEs. 
+
+=== BYNET 
+BYNET is a system interconnect that enables component communication. The BYNET possesses high-speed logic that provides bi-directional broadcast, multicast, and point-to-point communication and merge functions. It performs three key functions: coordinating multi-AMP queries, reading data from multiple AMPs while maintaining sorting, and regulating message flow to prevent congestion and protect platform throughput.
```

Comment:
```suggestion
BYNET is a system that enables component communication. The BYNET system provides high-speed bi-directional broadcast, multicast, and point-to-point communication and merge functions. It performs three key functions: coordinating multi-AMP queries, reading data from multiple AMPs, regulate message flow to prevent congestion and process platform throughput.